Add tests reflecting #19390

Lewis65 opened the pull request:
Added two tests to prevent checking a single property and passing the challenge.

The tests check two names at a time rather than adding 4 individual tests to the challenge.

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #19382
Closes #17734

Lewis65 committed:
Add test reflecting #19390

Added two tests to prevent checking only `Alan`.

Lewis65 commented:
First PR, so if I am doing something wrong please let me know :)

ezioda004 commented:
Hi @Lewis65,
Waiting for the conformation from the core team to accept PR for that bug.
Meanwhile I've notice you've introduced 2 tests, though it can still be passed by doing something like:
```js
return obj.hasOwnProperty('Alan') && obj.hasOwnProperty('Jeff');
```
I'd suggest adding tests for individual Properties, that way all would need to be present.

ezioda004 commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/check-if-an-object-has-a-property.english.md:
```diff
@@ -26,7 +26,10 @@ tests:
   - text: 'The function <code>isEveryoneHere</code> returns <code>false</code> if <code>Alan</code>, <code>Jeff</code>, <code>Sarah</code>, and <code>Ryan</code> are not properties on the <code>users</code> object'
     testString: 'assert((function() { delete users.Alan; delete users.Jeff; delete users.Sarah; delete users.Ryan; return isEveryoneHere(users) })() === false, "The function <code>isEveryoneHere</code> returns <code>false</code> if <code>Alan</code>, <code>Jeff</code>, <code>Sarah</code>, and <code>Ryan</code> are not properties on the <code>users</code> object");'
-
+    - text: 'The function <code>isEveryoneHere</code> returns <code>false</code> if <code>Alan</code>, <code>Jeff</code>, <code>Sarah</code>, and <code>Ryan</code> are not properties on the <code>users</code> object'
```
Can you also make sure the indentation of the `text` is same as above?

Lewis65 committed:
Split into 4 tests for each prop

Split previous tests with 2 props on each into 4 single-prop tests; fixed whitespace

nikrb commented:
@ezioda004 tested locally. 
confidence is low, but I don't think this challenge is working yet as it passes with just `return 'Alan' in obj;` as the solution. I think the problem is that the new tests remove users but don't add them back in, so Alan is always missing.
Apologies if I'm being too pedantic or have missed something.

ezioda004 commented:
Hi @nikrb,

Thanks for QA'ing. You are correct, after deleting `users.Alan`, its never added back and the subsequent tests returns `false`.

@Lewis65 Could you add the keys back in the next tests?
Example:
```js
(function() { delete users.Jeff; users.Alan = {age: 27, online: true}; return isEveryoneHere(users) })() === false
```
Likewise for other tests.

ValeraS commented:
@ezioda004 @nikrb For now, JS challenges execute tests independently. No need to restore initial data anymore.

nikrb commented:
tested this locally on latest master. curriculum test passes and client app looks good.

RandellDawson commented:
@ezioda004 Can you review this PR again to see if everything meets the requirement to close issue #19382?

RandellDawson reviewed (approved):
I have tested this PR locally and conclude it now fixed the bug described in the two issues #17734 and #19382. Great job @Lewis65!

ezioda004 commented:
Thank you @Lewis65 for your contribution and congratulations on your first PR merge to this repo! 🎉
Hope to see more contributions from you in the future.
